Production/test/unitTest.py

```diff
@@ -1,92 +1,157 @@
+    mytests.append(Test("Summer16sig","SMS-T1tttt_mGluino-1500_mLSP-100_TuneCUETP8M1_13TeV-madgraphMLM-pythia8",numevents,command,dataset="/store/mc/RunIISummer16MiniAODv2/SMS-T1tttt_mGluino-1500_mLSP-100_TuneCUETP8M1_13TeV-madgraphMLM-pythia8/MINIAODSIM/PUMoriond17_80X_mcRun2_asymptotic_2016_TrancheIV_v6-v1/50000/0818BABF-64BE-E611-B201-008CFA000280.root"))
+    mytests.append(Test("Autumn18","Autumn18.GJets_HT-600ToInf_TuneCP5_13TeV-madgraphMLM-pythia8_ext1",numevents,command,inputFilesConfig="Autumn18.GJets_HT-600ToInf_TuneCP5_13TeV-madgraphMLM-pythia8_ext1",nstart=0,nfiles=10))
+    mytests.append(Test("Autumn18Fast","Autumn18Fast.TTJets_SingleLeptFromT_TuneCP2_13TeV-madgraphMLM-pythia8",numevents,command,inputFilesConfig="Autumn18Fast.TTJets_SingleLeptFromT_TuneCP2_13TeV-madgraphMLM-pythia8",nstart=0,nfiles=1))
+    mytests.append(Test("Autumn18Fastsig","Autumn18Fast.SMS-T1tttt_TuneCP2_13TeV-madgraphMLM-pythia8",numevents,command,inputFilesConfig="Autumn18Fast.SMS-T1tttt_TuneCP2_13TeV-madgraphMLM-pythia8",nstart=0,nfiles=1))
+    mytests.append(Test("2018B26Sep",name,numevents,command,inputFilesConfig="Run2018B-26Sep2018-v1.JetHT",nstart=0,nfiles=10))
+    mytests.append(Test("2018B26SepHEM",name,numevents,command,inputFilesConfig="Run2018B-26Sep2018_HEM-v1.JetHT",nstart=0,nfiles=10))
+    mytests.append(Test("2018B26SepHEM",name,numevents,command,inputFilesConfig="Run2018B-26Sep2018_HEMmitigation-v1.JetHT",nstart=0,nfiles=10))
+    mytests.append(Test("2018PromptReco",name,numevents,command,inputFilesConfig="Run2018D-PromptReco-v2.JetHT",nstart=244,nfiles=10))
+    mytests.append(Test("2018ReReco17Sep",name,numevents,command,inputFilesConfig="Run2018B-17Sep2018-v1.JetHT",nstart=0,nfiles=10))
+
+    # make sure this is the last test defined
+    mytests.append(Test(scenario,name,numevents,command,dataset,inputFilesConfig,nstart=0,nfiles=10,redir=""))
+
+def unitTest():
+    # Read parameters
+    from TreeMaker.Utils.CommandLineParams import CommandLineParams
+    parameters = CommandLineParams()
+    fork=parameters.value("fork",True)
+    log=parameters.value("log",True)
+    clean=parameters.value("clean",0)
+    test=parameters.value("test",-1)
+    name=parameters.value("name","")
+    run=parameters.value("run",False)
+    numevents=parameters.value("numevents",100)
+    command=parameters.value("command","")
+    scenario=parameters.value("scenario","")
+    dataset=parameters.value("dataset","")
+    inputFilesConfig=parameters.value("inputFilesConfig","")
+
+    # only set name for a specific test
+    if test==-1: name = ""
+
+    # list of tests
+    mytests = []
+    defineTests(mytests,scenario,name,numevents,command,dataset,inputFilesConfig)
+
+    # sanity check for defining an on-the-fly test
+    if test==len(mytests) and dataset=="" and inputFilesConfig=="":
+        print "If defining a unitTest on-the-fly, you must either specify a \'dataset\' or an \'inputFilesConfig\'."
+        return -1
```

Comment:
this return seems to be discarded, should it be emitted as an exit code?

Reply:
Fixed